### PR TITLE
feat: add annotate SLO action

### PR DESCRIPTION
**Pull request by matkaras**
## Summary

Adds a GitHub action that creates a new [Nobl9 SLO Annotation](https://docs.nobl9.com/Features/SLO_Annotations). It is useful for annotating SLOs with metadata such as release versions, maintenance or other information that can be used to correlate SLOs with other data sources or events.

## Inputs

| Parameter                | Description                                               | Required | Default                    |
|--------------------------|-----------------------------------------------------------|----------|----------------------------|
| `annotation`             | Annotation kind description                               | **Yes**  | N/A                        |
| `project`                | Project containing SLO to annotate                        | **Yes**  | N/A                        |
| `nobl9_client_id`        | The Client ID of your Nobl9 account                       | **Yes**  | N/A                        |
| `nobl9_client_secret`    | The Client Secret of your Nobl9 account                   | **Yes**  | N/A                        |
| `labels`                 | Comma separated list of labels to filter SLOs to annotate | No       | N/A                        |
| `slo`                    | Name of single SLO to annotate                            | No       | N/A                        |
| `nobl9_okta_auth_server` | Nobl9 Okta auth server                                    | No       | https://accounts.nobl9.com |
| `nobl9_okta_org_url`     | Nobl9 Okta Organization URL                               | No       | auseg9kiegWKEtJZC416       |
| `nobl9_url`              | Nobl9 API URL                                             | No       | https://apps.nobl9.com/api |
| `sloctl_version`         | `sloctl` version used by the GitHub Action                | No       | v0.0.99                    |

> When no `labels` or `slo` are provided, all SLOs in the project will be annotated.
### Annotate all SLOs within a project
```yaml
name: Nobl9 Annotate SLO GitHub Actions Demo
on: [push]
jobs:
  annotate-slos:
    runs-on: ubuntu-latest
    steps:
      - uses: nobl9/action-annotate-slo@v0.1.0
        with:
          nobl9_client_id: ${{ secrets.CLIENT_ID }}
          nobl9_client_secret: ${{ secrets.CLIENT_SECRET }}
          annotation: "Release v1.0.0"
          project: "default"
```

### Annotate SLOs with labels
```yaml
name: Nobl9 Annotate SLO GitHub Actions Demo
on: [push]
jobs:
  annotate-slos:
    runs-on: ubuntu-latest
    steps:
      - uses: nobl9/action-annotate-slo@v0.1.0
        with:
          nobl9_client_id: ${{ secrets.CLIENT_ID }}
          nobl9_client_secret: ${{ secrets.CLIENT_SECRET }}
          annotation: "Release v1.0.0"
          project: "default"
          labels: "area=latency,team=orange,component=api"
```

### Annotate single SLO
```yaml
name: Nobl9 Annotate SLO GitHub Actions Demo
on: [push]
jobs:
  annotate-slos:
    runs-on: ubuntu-latest
    steps:
      - uses: nobl9/action-annotate-slo@v0.1.0
        with:
          nobl9_client_id: ${{ secrets.CLIENT_ID }}
          nobl9_client_secret: ${{ secrets.CLIENT_SECRET }}
          annotation: "Release v1.0.0"
          project: "default"
          slo: "my-slo-name"
```

### CODE REVIEW

File: post-annotations.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  annotation_name=$(printf "%.${max_chars}s-%s" "$slo_name" "$timestamp")
+
+  cat annotation.yaml.template >> annotations.yaml
+  sed -i"" -e "s/\$ANNOTATION_NAME/$annotation_name/g" \
+    -e "s/\$ANNOTATION_DESCRIPTION/$annotation/g" \
+    -e "s/\$PROJECT/$project/g" \
+    -e "s/\$SLO_NAME/$slo_name/g" \
+    -e "s/\$ANNOTATION_TIME/$annotation_time/g" \
+    -e "s/\$ANNOTATION_TIME/$annotation_time/g" \
+    annotations.yaml
```
Comment (triluch) [inline post-annotations.sh]: `envsusbst` might be safer alternative here than `sed`, however even that isn't really super safe (no escaping). Maybe something like described here: https://stackoverflow.com/questions/70713452/yq-replace-value-in-manifest-yaml to handle escaping properly (using `strenv()` should work), as you already have yq here.

File: action.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      id: annotate-slo
+      env:
+        SLOCTL_CLIENT_ID: ${{ inputs.nobl9_client_id }}
+        SLOCTL_CLIENT_SECRET: ${{ inputs.nobl9_client_secret }}
+        SLOCTL_OKTA_ORG_URL: ${{ inputs.nobl9_okta_org_url }}
+        SLOCTL_OKTA_AUTH_SERVER: ${{ inputs.nobl9_okta_auth_server }}
+        SLOCTL_URL: ${{ inputs.nobl9_url }}
+      run: |
+        exec ${{ github.action_path }}/post-annotations.sh \
+        -p "${{ inputs.project }}" \
```
Comment (triluch) [inline action.yaml]: As those are used as parameters I'd put some validation on them.

**Review by altar64 — APPROVED** [review]
👍🏼

File: post-annotations.sh (hunk 1 above)
Comment (matkaras) [reply post-annotations.sh]: changed to yq

File: action.yaml (hunk 2 above)
Comment (matkaras) [reply action.yaml]: added regex validation to project, slo, and labels

**Review by triluch — APPROVED** [review]

**Pull request merged**